packages/account/src/providers/provider.test.ts

```diff
@@ -209,7 +202,7 @@ describe('Provider', () => {
     const providerUrl2 = 'http://127.0.0.1:8080/graphql';
 
     const provider = await Provider.create(providerUrl1, {
-      fetch: (url: string, options: FetchRequestOptions) =>
+      fetch: (url, options) =>
```

Comment:
Same thing about argument's type.

Reply:
I can see the benefit in adding the type annotations for the docs snippets, but here I prefer leaving it like this as it's an internal test and type inference would warn us if something was wrong.

Reply:
The downside of needlessly relying on type inference is that it makes it harder for the next person to know the type since one can't simply _read_ the code to figure it out. This requires some extra pirouettes, like hovering the variable, and we don't have a type definition to click and go to.

What's the benefit of purposefully not adding the type notation?

Reply:
Whoever wants to see the type of `options` can F12 into the `fetch` property and it'll take them directly to the function definition. 

The benefit is less code; we'd be repeating the type of the parameter while it's already defined and inferred in the body of the function. 
Also, doing any type annotation on a function's parameters that can be inferred can lead to a mismatch between the annotated type and the actual underlying type, which'd give compilation errors even though the function's implementation isn't even affected by the change (e.g. adding an additional property to the underlying type that's missing in the annotated type). The only way to _properly_ annotate in these cases and ensure that no compilation mismatch happens is like this:
```ts
{
      fetch: (...[url, options]: Parameters<NonNullable<ProviderOptions['fetch']>>) =>
       getCustomFetch('getVersion', { nodeInfo: { nodeVersion: url } })(url, options),
}
```
It looks silly, but it's the end conclusion of proper type annotation that doesn't give unnecessary compilation errors in the body of the function. One can say that it's easy to change the simple non-silly annotation to make TS happy, but I prefer TS not to interfere when there's no need to interfere (e.g. when the body of the function isn't affected).

In the end, it's a matter of style. I like my TS code inferred based on a single source of truth and not make any assumptions (annotations) in the source's consumers. If the consumer is incompatible with the source type, TS will yell.

Reply:
Less code, in this case, is irrelevant.

The docs exemplify the use of type annotation, and so should the source.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1822/commits/c7558749b792d935c5342badb1239fb8ff3035f0